Try to make flow control consuming and processing crisper (#299)
quicwg/ops-drafts: draft-ietf-quic-applicability.md

provide flow control credit.  Understanding what causes deadlocking might help
implementations avoid deadlocks.

> Applications that use QUIC often have a data consumer that reads data from
> transport buffers. Timely updates of transport flow control credit can improve
> performance. A common flow control implementation technique is for a QUIC

Reviewer: Reorder these first two sentences:

```suggestion
Timely updates of transport flow control credit can improve performance.
Applications that use QUIC often have a data consumer that reads data from
transport buffers. A common flow control implementation technique is for a QUIC
```

receiver to extend credit to the sender as the data consumer reads data.  Some
implementations might have independent transport-layer and application-layer
receive buffers. Consuming data does not always imply it is immediately

> processed. However, when data is consumed,

Reviewer: Incomplete sentence.

provide flow control credit.  Understanding what causes deadlocking might help
implementations avoid deadlocks.

The size and rate of transport flow control credit updates can affect
performance. Applications that use QUIC often have a data consumer that reads
data from transport buffers. Some implementations might have independent
transport-layer and application-layer receive buffers. Consuming data does not
always imply it is immediately processed. However, a common flow control
implementation technique is to extend credit to the sender, by emitting MAX_DATA
and/or MAX_STREAM_DATA frames, as data is consumed. Delivery of these frames

> is affected by uplink latency. If credit is not extended in a timely manner, the

Reviewer 2: ```suggestion
is affected by the latency of the back channel from the receiver to the data sender.
If credit is not extended in a timely manner, the
```
Is this any better?

provide flow control credit.  Understanding what causes deadlocking might help
implementations avoid deadlocks.

The size and rate of transport flow control credit updates can affect
performance. Applications that use QUIC often have a data consumer that reads
data from transport buffers. Some implementations might have independent
transport-layer and application-layer receive buffers. Consuming data does not
always imply it is immediately processed. However, a common flow control
implementation technique is to extend credit to the sender, by emitting MAX_DATA
and/or MAX_STREAM_DATA frames, as data is consumed. Delivery of these frames
is affected by the latency of the back channel from the receiver to the data sender.
If credit is not extended in a timely manner, the

> sending application can be blocked, effectively throttling the downlink.

Reviewer 3: ```suggestion
sending application can be blocked, effectively throttling the connection.
```
